### PR TITLE
[WIP] AWS sdk upgrade

### PR DESCRIPTION
**Pull request by junyuc25**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -156,8 +156,10 @@
     <!-- Should be consistent with SparkBuild.scala and docs -->
     <avro.version>1.11.1</avro.version>
     <aws.kinesis.client.version>1.12.0</aws.kinesis.client.version>
+    <aws.kinesis.client.v2.version>2.4.3</aws.kinesis.client.v2.version>
     <!-- Should be consistent with Kinesis client dependency -->
-    <aws.java.sdk.version>1.11.655</aws.java.sdk.version>
+    <aws.java.sdk.version>1.12.447</aws.java.sdk.version>
+    <aws.java.sdk.v2.version>2.17.268</aws.java.sdk.v2.version>
```
Comment (dongjoon-hyun) [inline pom.xml:162]: Is this PR aiming to upgrade both V1 and V2 at the same time?

**Comment by pan3793** [comment]
Is it possible to provide dedicated credentials for the Spark community to run CI with AWS services?

File: project/plugins.sbt [hunk 2]
```diff
@@ -46,3 +46,5 @@ addSbtPlugin("com.simplytyped" % "sbt-antlr4" % "0.8.3")
 addSbtPlugin("com.github.sbt" % "sbt-pom-reader" % "2.4.0")
 
 addSbtPlugin("com.thesamet" % "sbt-protoc" % "1.0.6")
+
+addDependencyTreePlugin
```
Comment (LuciferYang) [inline project/plugins.sbt:50]: This change will eventually be revert, right?

**Comment by LuciferYang** [comment]
I have two concerns about continuing to maintain this module:
1. This module has 60 UTs, but 24 are ignored due to the environment variable `ENABLE_KINESIS_TESTS` defaulting to 0. 
```
[info] Tests: succeeded 34, failed 0, canceled 0, ignored 24, pending 0
[info] All tests passed.
```
However, when I run `ENABLE_KINESIS_TESTS=1 build/sbt "streaming-kinesis-asl/test" -Pkinesis-asl`, these previously ignored test cases fail.
```
[info] *** 4 SUITES ABORTED ***
[error] Error: Total 36, Failed 0, Errors 4, Passed 32
[error] Error during tests:
[error] 	org.apache.spark.streaming.kinesis.WithoutAggregationKinesisBackedBlockRDDSuite
[error] 	org.apache.spark.streaming.kinesis.WithAggregationKinesisBackedBlockRDDSuite
[error] 	org.apache.spark.streaming.kinesis.WithAggregationKinesisStreamSuite
[error] 	org.apache.spark.streaming.kinesis.WithoutAggregationKinesisStreamSuite
[error] (streaming-kinesis-asl / Test / test) sbt.TestsFailedException: Tests unsuccessful
```
2. The CI does not provide sufficient check for this module.

https://github.com/apache/spark/blob/a21e19b6e7ac4c4c77b39d93a2da2cbe1c88c4c8/.github/workflows/build_and_test.yml#L139

So, I think we should complete the TODO SPARK-32246 before proceeding with this update.

**Pull request closed without merging**